hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystem.java

```diff
@@ -6537,14 +6537,45 @@ public String getLiveNodes() {
       if (node.getUpgradeDomain() != null) {
         innerinfo.put("upgradeDomain", node.getUpgradeDomain());
       }
+      StorageReport[] storageReports = node.getStorageReports();
+      innerinfo.put("blockPoolUsedPercentStdDev",
+          getBlockPoolUsedPercentStdDev(storageReports));
       info.put(node.getXferAddrWithHostname(), innerinfo.build());
     }
     return JSON.toString(info);
   }
 
+  /**
+   * Return the standard deviation of storage block pool usage.
+   */
+  @VisibleForTesting
+  public float getBlockPoolUsedPercentStdDev(StorageReport[] storageReports) {
+    ArrayList<Float> usagePercentList = new ArrayList<>();
+    float totalUsagePercent = 0.0f;
+    float dev = 0.0f;
+
+    if (storageReports.length == 0) {
+      return dev;
+    }
+
+    for (StorageReport s : storageReports) {
+      usagePercentList.add(s.getBlockPoolUsagePercent());
+      totalUsagePercent += s.getBlockPoolUsagePercent();
+    }
+
+    totalUsagePercent /= storageReports.length;
+    Collections.sort(usagePercentList);
```

Comment:
Unnecessary sort

Reply:
Thank you for pointing that. I will fix it.

Reply:
@ferhui A float or double may lose precision when being evaluated. When multiple values are operated on in different order, the results may be inconsistent. After remote ```Collections.sort(usagePercentList);```, I only take two decimal points to assert. Please take a look at this. Thank you very much.

Reply:
Thank @ferhui for your comments, I will try this. At present, there is a failed UT, which is related to my changes, and I am repairing it. In order not to delay your time, you can review it after I fix it.